src/corefx/System.Globalization.Native/pal_locale.c

```diff
@@ -156,7 +182,26 @@ const char* DetectDefaultLocaleName()
         }
     }
 
-    return uloc_getDefault();
+    // Check if defined default locale is "C" and then map it to invariant instead of posix locale
```

Comment:
Why is this logic so complex? I would expect the whole thing to be just:

```
const char* DetectDefaultLocaleName()
{
    const char* icuLocale = uloc_getDefault();
    if (strcmp(icuLocale, "en_US_POSIX") == 0)
    {
        return "";
    }
}
```

Reply:
because we need to honor anyone setting LANG to posix. something like 

```
LANG=en_US_POSIX
```

we shouldn't change this to invariant because it was intentional setting exact needed culture.